### PR TITLE
`--api-timeout` flag

### LINKED ISSUE #9917: Make API timeout configurable
# 🚀 Feature Request

### Description

It is currently not possible to specify an HTTP timeout for API requests. The value is hard-coded at 2 minutes here: https://github.com/prysmaticlabs/prysm/blob/a78cdf86cc4e614ced18aa7af32c89b4372657c7/api/gateway/apimiddleware/process_request.go#L80
This is problematic because some queries might take longer, especially those which access past states.

### Describe the solution you'd like

Add a CMD `--api-timeout` flag that accepts the desired timeout value in minutes. It should default to `2`.

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Allows to specify a timeout value for API requests via a flag.

**Which issues(s) does this PR fix?**

Fixes #9917

**Other notes for review**

Supersedes https://github.com/prysmaticlabs/prysm/pull/9946

### CODE REVIEW

File: cmd/flags.go [hunk 1]
```diff
@@ -249,6 +250,12 @@ var (
 		Usage: "Specifies the size in bytes of bolt db's mmap syscall allocation",
 		Value: 536870912, // 512 Mb as a default value.
 	}
+	// ApiTimeoutFlag specifies the timeout value for API requests in minutes. A timeout of zero means no timeout.
+	ApiTimeoutFlag = &cli.IntFlag{
+		Name:  "api-timeout",
+		Usage: "Specifies the timeout value for API requests in minutes",
```
Comment (potuz) [inline cmd/flags.go:256]: do users want minutes or seconds here? I can imagine a situation where users may want to use this flag to restrict calls to be less than a minute or fail

**Review by potuz — COMMENTED** [review]
It looks good to me

File: cmd/flags.go [hunk 2]
```diff
@@ -249,6 +250,12 @@ var (
...
+		Value: int(2 * time.Minute),
```
Comment (prestonvanloon) [inline cmd/flags.go:257]: You are asking the user to put time in minutes, but here you have the time in nanoseconds.

Comment (potuz) [reply cmd/flags.go:257]: I think reverting 5da5489 would bring back minutes as it was the only change that commit brought. Still I wonder if it's not a valid usecase to have timeouts lower than a minute

File: cmd/flags.go [hunk 3]
```diff
@@ -249,6 +250,12 @@ var (
 		Usage: "Specifies the size in bytes of bolt db's mmap syscall allocation",
 		Value: 536870912, // 512 Mb as a default value.
 	}
+	// ApiTimeoutFlag specifies the timeout value for API requests in minutes. A timeout of zero means no timeout.
+	ApiTimeoutFlag = &cli.IntFlag{
```
[the remark this answers is not in the document]
Comment (rkapka) [reply cmd/flags.go:253]: I think simply having integers as seconds is enough for us, we don't need more flexibility. A duration flag requires users to pass values that will get properly parsed e.g. `10s`.

**Review by potuz — APPROVED** [review]
I can't find anything wrong with it

**Comment by rkapka** [comment]
I think Preston had a valid comment, let's not merge it yet

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**